lib/warden/strategies/pin_strategy.rb

```diff
@@ -4,23 +4,37 @@ module Warden
   module Strategies
     class PinStrategy < Warden::Strategies::Base
       def valid?
-        params["pin"].present? && params['otp_auth_key'].present?
+        request_params[:pin].present? && request_params[:otp_auth_key].present?
       end
 
-      def authenticate!
-        auth_token = AuthToken.find_by_otp_auth_key(params['otp_auth_key'])
+      def lookup_auth_token
+        AuthToken.find_by_otp_auth_key(request_params[:otp_auth_key])
+      end
+
+      def pin_method
+        request_params[:pin_for]&.to_sym || :mobile
+      end
+
+      def authenticate!        
+        auth_token = lookup_auth_token
         return success!(auth_token.user) if valid_app_store_credentials?(auth_token)
         user = auth_token.try(:user)
-        has_valid_otp_code?(auth_token) && valid_user(user) ? success!(user) : fail
+        return fail unless has_valid_otp_code?(auth_token) && valid_user(user)
+        user.set_verified_flag(pin_method) if pin_method.present?
```

Comment:
don't we want to get the `pin_method` from the JWT token?

Reply:
@steveyken That's the V1 implementation, look at `pin_jwt_strategy.rb`

Reply:
I inherited this one, and made some changes